### PR TITLE
Change to using libyaml-dev, to speed up pyyaml

### PR DESCRIPTION
**Pull request by hashbrowncipher**
If we have to build our own wheel, we need libyaml-dev headers in order
to get C accelerated yaml parsing.

### CODE REVIEW

**Comment by solarkennedy** [comment]
Ship. I guess it is "ok" for s_c_l to be tolerant of missing C extensions, but this code and paasta code needs this.

**Comment by solarkennedy** [comment]
Is it easy to add an itest?
```
/usr/share/python/nerve-tools/bin/python -c 'import yaml; assert yaml.__with_libyaml__'
```

**Comment by jolynch** [comment]
@hashbrowncipher please do the same for synapse-tools if this solution works

File: dockerfiles/trusty/Dockerfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ MAINTAINER John Billings <billings@yelp.com>
 RUN apt-get update
 
 RUN apt-get -y install dpkg-dev python-tox python-setuptools
-RUN apt-get -y install python-dev debhelper libyaml-0-2 libcurl4-openssl-dev
+RUN apt-get -y install python-dev debhelper libyaml-dev libcurl4-openssl-dev
```
Comment (jolynch) [inline dockerfiles/trusty/Dockerfile]: So do we have to have the debian package explicitly depend on libyaml then?

Comment (solarkennedy) [inline dockerfiles/trusty/Dockerfile]: AFAIK, no thanks to [dh_makeshlibs](http://man.cx/dh_makeshlibs(1)) per @hashbrowncipher

Comment (jolynch) [inline dockerfiles/trusty/Dockerfile]: Alright then :-)

**Comment by hashbrowncipher** [comment]
Yeah, this deserves an itest.

**Pull request closed without merging**